### ISSUE TITLE
flask-admin can't be upgraded

### ISSUE THREAD

**Issue by backwardspy** [binkhq/api-reflector#70]
the latest version breaks our inline models: https://github.com/flask-admin/flask-admin/issues/1405
there are a few workarounds mentioned in the issue commits, we'll need to try those when we want to upgrade to the latest version.

**Comment by haffi96** [comment]
Did you notice this with 1.6.1? I haven't been able to reproduce any breakages after updating 🤔

**Comment by backwardspy** [comment]
this does indeed seem to be working again now!

**Comment by thespidey** [comment]
SQLAlchemy model mappings, specifically in the _calculate_mapping_key_pair function. This function is critical for determining the relationship mappings between our models. We observed that the direction attribute of SQLAlchemy model properties was not being updated as expected during runtime. This inconsistency was not apparent during debugging sessions when inspecting variables, leading to a challenging troubleshooting process.

Workaround Implemented:
After extensive testing and analysis, we found that invoking configure_mappers() from sqlalchemy.orm before executing the logic in the _calculate_mapping_key_pair function resolved the issue. This suggests that the mapper configurations were not being fully established or updated automatically during normal application flow.

This behavior seems to align with the challenges mentioned in the issue thread regarding inline models. The requirement of manually invoking configure_mappers() to ensure proper mapping configurations highlights a potential underlying issue in the integration between SQLAlchemy and the latest version of Flask-Admin.
